src/NuGet.Clients/PackageManagement.VisualStudio/IDE/IDeferredProjectWorkspaceService.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    [Guid("113bc32d-986a-4ef4-b5d8-f1e3da44c0f8")]
+    public interface IDeferredProjectWorkspaceService
+    {
+        Task<bool> EntityExists(string filePath);
+
+        Task<IEnumerable<string>> GetProjectReferencesAsync(string projectFilePath);
```

Comment:
Only for .Net core projects. Its good for packages.config or project.json projects. I'll communicate with DPL folks to provide TFM info as well with project references.